File: december 2017/bronze/billboard/billboard.cpp
/*
ID: warwick2
LANG: C++11
TASK: base
*/
#include <iostream>
#include <fstream>
#include <algorithm>
#include <set>
#include <list>
#include <cmath>
using namespace std;

struct rectangle{
    int left;
    int top;
    int right;
    int bottom;
};

int total;
rectangle truck;

void add_to_total(rectangle rect){
    int width = rect.right-rect.left;
    int height = rect.top-rect.bottom;
    total += width*height;
}

void subtract_from_total(rectangle rect){
    for (int x=rect.left; x<rect.right; x++){
        for (int y=rect.bottom; y<rect.top; y++){
            // Check if x in truck
            if (x >= truck.left && x < truck.right){
                // Check if y in truck
                if (y >= truck.bottom && y < truck.top){
                    total--;
                }
            }
        }
    }
}

int main(void){
	ifstream fin("billboard.in");
    ofstream fout("billboard.out");

    rectangle billboard0;  
    rectangle billboard1;

    fin >> billboard0.left >> billboard0.bottom >> billboard0.right >> billboard0.top;
    fin >> billboard1.left >> billboard1.bottom >> billboard1.right >> billboard1.top;
    fin >> truck.left >> truck.bottom >> truck.right >> truck.top;

    add_to_total(billboard0);
    add_to_total(billboard1);

    cout << total << "\n";

    subtract_from_total(billboard0);
    subtract_from_total(billboard1);

    cout << total << "\n";


    fout << total << "\n";
    return 0;
}

File: december 2017/bronze/billboard/billboard.py
"""
ID: warwick2
LANG: PYTHON3
TASK: billboard
"""
with open('billboard.in', 'r') as fin:
    in_text = fin.read().strip()
    lines = in_text.split('\n')

billboard0 = lines[0].split(' ')
billboard0 = list(map(lambda x: int(x), billboard0))
billboard1 = lines[1].split(' ')
billboard1 = list(map(lambda x: int(x), billboard1))
truck = lines[2].split(' ')
truck = list(map(lambda x: int(x), truck))

width0 = billboard0[2]-billboard0[0]
height0 = billboard0[3]-billboard0[1]
width1 = billboard1[2]-billboard1[0]
height1 = billboard1[3]-billboard1[1]

total = width0*height0 + width1*height1

print(total)

for x in range(billboard0[0], billboard0[2]):
    for y in range(billboard0[1], billboard0[3]):
        print((x, y))
        if x >= truck[0] and x < truck[2]:
            if y >= truck[1] and y < truck[3]:
                total -= 1

for x in range(billboard1[0], billboard1[2]):
    for y in range(billboard1[1], billboard1[3]):
        if x >= truck[0] and x < truck[2]:
            if y >= truck[1] and y < truck[3]:
                total -= 1

print(total)

out = str(total)
out += '\n'
with open('billboard.out', 'w') as fout:
    fout.write(out)